File: baekjoon/lecture/basic_algorithm/questions/540_brute_force/01_q11723.py
import sys

sys.stdin = open("/mnt/repository/coding-test/baekjoon/lecture/basic_algorithm/questions/input.txt", "r")

def command_processor() :
    m = int(sys.stdin.readline().strip())
    s = 0b0

    for _ in range(m):
        command = sys.stdin.readline().strip()

        try:
            op, x = command.split()
            x = int(x) - 1
            if op == 'add':
                s = s | (0b1<<x)
            elif op == 'remove':
                s = s & ~(0b1<<x)
            elif op == 'check':
                if (s & (0b1<<x)):
                    print(1)
                else:
                    print(0)
            elif op == 'toggle':
                s = s ^ (0b1<<x)
        except:
            if command == 'all':
                s = 0b11111111111111111111
            elif command == 'empty':
                s = 0b0

command_processor()

File: baekjoon/lecture/basic_algorithm/540_brute_force/01_q11723.js
const fs = require('fs');
const input = fs.readFileSync('../input.txt').toString().trim().split('\n');

const m = +input.shift();
const commands = input.map((command) => command.split(' '));

let s = new Array(20).fill(0);

commandProcessor(commands);

function commandProcessor(commands) {
  let op, x, idx;
  let result = '';
  commands.forEach((cmd) => {
    op = cmd[0];
    switch (op) {
      case 'add':
        x = cmd[1];
        if (!s[x - 1]) {
          s[x - 1] = 1;
        }
        break;
      case 'remove':
        x = cmd[1];
        if (s[x - 1]) {
          s[x - 1] = 0;
        }
        break;
      case 'check':
        x = cmd[1];
        if (s[x - 1]) {
          result += '1';
        } else {
          result += '0';
        }
        result += '\n';
        break;
      case 'toggle':
        x = cmd[1];
        if (s[x - 1]) {
          s[x - 1] = 0;
        } else {
          s[x - 1] = 1;
        }
        break;
      case 'all':
        s.fill(1);
        break;
      case 'empty':
        s.fill(0);
        break;
    }
  });
  console.log(result.trim());
}
